playground/infrastructure/helper.py

```diff
@@ -0,0 +1,171 @@
+    return examples
+
+
+async def get_statuses(examples: [Example]):
+    """ Receive status and update example.status and example.pipeline_id for each example
+
+    Args:
+        examples: beam examples for processing and updating statuses and pipeline_id values.
+    """
+    tasks = []
+    for example in examples:
+        tasks.append(_update_example_status(example))
+    await asyncio.gather(*tasks)
+
+
+def _get_example(filepath: str, filename: str) -> Example:
+    """ Return an Example by filepath and filename.
+
+    Args:
+         filepath: path of the example's file.
+         filename: name of the example's file.
+
+    Returns:
+        Return an Example.
+    """
+    name = _get_name(filename)
+    sdk = _get_sdk(filename)
+    with open(filepath) as parsed_file:
+        content = parsed_file.read()
+
+    return Example(name, "", sdk, filepath, content, "", STATUS_UNSPECIFIED)
+
+
+def _match_pattern(filepath: str) -> bool:
+    """Check file to matching
+
+    Check that file has the correct extension and contains the beam-playground tag.
+
+    Args:
+        filepath: path to the file.
+
+    Returns:
+        True if file matched. False if not
+    """
+    extension = filepath.split(os.extsep)[-1]
+    if extension in SUPPORTED_SDK:
+        with open(filepath) as parsed_file:
+            content = parsed_file.read()
+        return re.search(PATTERN, content) is not None
+
+
+def _get_name(filename) -> str:
+    """ Return name of the example by his filepath.
+
+    Get name of the example by his filename.
+
+    Args:
+        filename: filename of the beam example file.
+
+    Returns:
+        example's name.
+    """
+    return filename.split(os.extsep)[0]
+
+
+def _get_sdk(filename) -> Sdk:
+    """ Return SDK of example by his filename.
+
+    Get extension of the example's file and returns associated SDK.
+
+    Args:
+        filename: filename of the beam example.
+
+    Returns:
+        Sdk according to file extension.
+    """
+    extension = filename.split(os.extsep)[-1]
+    if extension in SUPPORTED_SDK:
+        return SUPPORTED_SDK[extension]
+    else:
+        raise ValueError(extension + " is not supported now")
+
+
+async def _update_example_status(example: Example):
+    """ Receive status for examples and update example.status and pipeline_id
+
+    Use client to send requests to the backend:
+    1. Start code processing.
+    2. Ping the backend while status is STATUS_VALIDATING/STATUS_PREPARING/STATUS_COMPILING/STATUS_EXECUTING
+    Update example.status with resulting status.
+
+    Args:
+        example: beam example for processing and updating status and pipeline_id.
+    """
+    client = GRPCClient()
+    pipeline_id = await client.run_code(example.code, example.sdk)
+    example.pipeline_id = pipeline_id
+    status = await client.check_status(pipeline_id)
+    while status in [STATUS_VALIDATING, STATUS_PREPARING, STATUS_COMPILING, STATUS_EXECUTING]:
+        await asyncio.sleep(1)
```

Comment:
I would sleep for more tan 1s. Beam pipelines especially on direct runner take some time. 10 or even 20 seconds is better.

Reply:
Changed.